### PR TITLE
MobilenetV2

### PR DESCRIPTION
**Pull request by HalfdanJ**
Add support for mobilenetV2. Set as default model, but mobilenetV1 can still be used using the new modelOptions argument. 

Adds karma test testing both v1 and v2.

### CODE REVIEW

**Comment by HalfdanJ** [comment]
Added changes that uses the last layer of mobilenetv2, and adds a globalAveragePooling2d as recommended by tf.js team.

File: image/src/mobilenet/custom-mobilenet.ts [hunk 1]
```diff
@@ -184,13 +204,18 @@ export class CustomMobileNet {
...
- * @param checkpoint the URL to load the mobilenet json
+ * @param modelOptions options determining what model to load
  */
-export async function loadTruncatedMobileNet(checkpoint: string = MOBILENET_MODEL_PATH) {
-    const mobilenet = await tf.loadLayersModel(checkpoint);
-    const layer = mobilenet.getLayer('conv_pw_13_relu');
-    const truncatedModel = tf.model({ inputs: mobilenet.inputs, outputs: layer.output });
-    return truncatedModel;
+export async function loadTruncatedMobileNet(modelOptions?: ModelOptions) {
+    const [checkpointUrl, trainingLayer] = parseModelOptions(modelOptions);
```
Comment (hapticdata) [inline image/src/mobilenet/custom-mobilenet.ts]: `trainingLayer` appears to never be used? and is included in `ModelOptions`, here in this function and in `parseModelOptions` is it used somewhere and I am missing it?

File: image/src/mobilenet/teachable-mobilenet.ts [hunk 2]
```diff
@@ -48,7 +49,7 @@ function flatOneHot(label: number, numClasses: number) {
 function convertToTfDataset(xs: Float32Array[], ys: number[][]) {
     const xTrain = tf.data.array(xs);
     const yTrain = tf.data.array(ys);
-
+
```
Comment (hapticdata) [inline image/src/mobilenet/teachable-mobilenet.ts]: mind removing these whitespaces, the linter will pick them up

File: image/src/mobilenet/teachable-mobilenet.ts [hunk 3]
```diff
@@ -108,12 +109,11 @@ export class TeachableMobileNet extends CustomMobileNet {
      * @param sample the image / tensor that belongs in this classification
      */
     // public async addExample(className: number, sample: HTMLCanvasElement | tf.Tensor) {
-    public async addExample(className: number, sample: HTMLCanvasElement | tf.Tensor) {
+    public async addExample(className: number, sample: HTMLImageElement | HTMLCanvasElement | tf.Tensor) {
         const cap = isTensor(sample) ? sample : capture(sample);
         const example = this.truncatedModel.predict(cap) as tf.Tensor;
-
+
```
Comment (hapticdata) [inline image/src/mobilenet/teachable-mobilenet.ts]: mind removing these whitespaces, the linter will pick them up

File: image/src/mobilenet/custom-mobilenet.ts [hunk 4]
```diff
@@ -207,7 +203,7 @@ export class CustomMobileNet {
  * @param modelOptions options determining what model to load
  */
 export async function loadTruncatedMobileNet(modelOptions?: ModelOptions) {
-    const [checkpointUrl, trainingLayer] = parseModelOptions(modelOptions);
+    const checkpointUrl = parseModelOptions(modelOptions);
```
Comment (hapticdata) [inline image/src/mobilenet/custom-mobilenet.ts]: does it make sense for this to really have an interface at this point? It only can be a `checkpointURL: string` or a `alpha: number` and never both and is only used in this method. How about:

```ts
export async function loadTruncatedMobileNet(checkpointURL_or_alpha: string | number) {
    if (typeof checkpointURL_or_alpha === 'string') {...
```

Comment (hapticdata) [reply image/src/mobilenet/custom-mobilenet.ts]: after discussing, I agree, maintaining `ModelOptions` will set us up better for future possibilities

**Pull request merged**